hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -208,71 +155,85 @@ private HoodieLogBlock readBlock() throws IOException {
     long blockEndPos = inputStream.getPos();
 
     switch (Objects.requireNonNull(blockType)) {
-      // based on type read the block
       case AVRO_DATA_BLOCK:
         if (nextBlockVersion.getVersion() == HoodieLogFormatVersion.DEFAULT_VERSION) {
-          return HoodieAvroDataBlock.getBlock(content, readerSchema);
+          return HoodieAvroDataBlock.getBlock(content.get(), readerSchema);
         } else {
-          return new HoodieAvroDataBlock(logFile, inputStream, Option.ofNullable(content), readBlockLazily,
-              contentPosition, contentLength, blockEndPos, readerSchema, header, footer, keyField);
+          return new HoodieAvroDataBlock(logFile, inputStream, content, readBlockLazily,
+              contentPosition, contentLength, blockEndPos, Option.ofNullable(readerSchema), header, footer, keyField);
         }
+
       case HFILE_DATA_BLOCK:
-        return new HoodieHFileDataBlock(logFile, inputStream, Option.ofNullable(content), readBlockLazily,
-            contentPosition, contentLength, blockEndPos, readerSchema,
-            header, footer, enableInlineReading, keyField);
+        checkState(nextBlockVersion.getVersion() != HoodieLogFormatVersion.DEFAULT_VERSION,
+            String.format("HFile block could not be of version (%d)", HoodieLogFormatVersion.DEFAULT_VERSION));
+
+        return new HoodieHFileDataBlock(logFile, inputStream, content, readBlockLazily,
+            contentPosition, contentLength, blockEndPos, Option.ofNullable(readerSchema),
+            header, footer, keyField, enableRecordLookups);
+
+      case PARQUET_DATA_BLOCK:
+        checkState(nextBlockVersion.getVersion() != HoodieLogFormatVersion.DEFAULT_VERSION,
+            String.format("Parquet block could not be of version (%d)", HoodieLogFormatVersion.DEFAULT_VERSION));
+
+        return new HoodieParquetDataBlock(logFile, inputStream, content, readBlockLazily,
+            contentPosition, contentLength, blockEndPos, Option.ofNullable(readerSchema), header, footer, keyField);
+
       case DELETE_BLOCK:
-        return HoodieDeleteBlock.getBlock(logFile, inputStream, Option.ofNullable(content), readBlockLazily,
+        return HoodieDeleteBlock.getBlock(logFile, inputStream, content, readBlockLazily,
             contentPosition, contentLength, blockEndPos, header, footer);
+
       case COMMAND_BLOCK:
-        return HoodieCommandBlock.getBlock(logFile, inputStream, Option.ofNullable(content), readBlockLazily,
+        return HoodieCommandBlock.getBlock(logFile, inputStream, content, readBlockLazily,
             contentPosition, contentLength, blockEndPos, header, footer);
+
       default:
         throw new HoodieNotSupportedException("Unsupported Block " + blockType);
     }
   }
 
+  @Nullable
+  private HoodieLogBlockType tryReadBlockType(HoodieLogFormat.LogFormatVersion blockVersion) throws IOException {
+    if (blockVersion.getVersion() == HoodieLogFormatVersion.DEFAULT_VERSION) {
+      return null;
```

Comment:
While addressing this i've stumbled upon peculiar behavior that is currently implemented: for `DEFAULT_VERSION` block reading would actually fail since it doesn't contain block-type, and we assert at line 252 (before change) that it's not null. 

While this doesn't seem right, i don't thin i have a good idea around what should be the behavior since we can't reliably determine what block is in-front of us and it seems that DEFAULT_VERSION supported Avro, Delete, Corrupt blocks